app/models/sign_in/o_auth_session.rb

```diff
@@ -20,7 +20,7 @@ def active?
     end
 
     def user_attributes_hash
-      @user_attributes_hash ||= JSON.parse(user_attributes)
+      @user_attributes_hash ||= user_attributes && JSON.parse(user_attributes)
```

Comment:
Would this be better? `@user_attributes_hash ||= JSON.parse(user_attributes) if user_attributes.present?`

Reply:
It might make sense to use a ternary to assign an empty hash which lets us avoid any memoization wonkiness. Otherwise, `user_attributes_hash` would be `nil`, resulting in memoization not working as expected.

```ruby
def user_attributes_hash
  @user_attributes_hash ||= user_attributes.present? ? JSON.parse(user_attributes) : {}
end
```

Reply:
Also, current implementation will return `false` if `user_attributes` is missing, also defeating memoization. Returning an empty hash would fix memoization and also eliminate the need for downstream nil checks.